aws/alarms/cloudwatch.tf

```diff
@@ -453,6 +453,38 @@ resource "aws_cloudwatch_metric_alarm" "generate_temporary_token_api_failure" {
   alarm_actions     = [var.sns_topic_alert_warning_arn]
   ok_actions        = [var.sns_topic_alert_ok_arn]
 
+  tags = {
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+resource "aws_cloudwatch_log_metric_filter" "request_temporary_token_api_using_unauthorized_email_address" {
+  name           = "RequestTemporaryTokenApiUsingUnauthorizedEmailAddress"
+  pattern        = "An email address with no access to any form has been locked out"
+  log_group_name = var.ecs_cloudwatch_log_group_name
+  metric_transformation {
+    name      = "RequestTemporaryTokenApiUsingUnauthorizedEmailAddress"
+    namespace = "forms"
+    value     = "1"
+  }
+}
+
+resource "aws_cloudwatch_metric_alarm" "request_temporary_token_api_using_unauthorized_email_address" {
+  alarm_name          = "RequestTemporaryTokenApiUsingUnauthorizedEmailAddress"
+  namespace           = "forms"
+  metric_name         = aws_cloudwatch_log_metric_filter.request_temporary_token_api_using_unauthorized_email_address.metric_transformation[0].name
```

Comment:
This should probably be `aws_cloudwatch_log_metric_filter.request_temporary_token_api_using_unauthorized_email_address.name`

Reply:
That also works but you have to make sure both the metric filter and the metric resulting from that filter have the same name. This is why I have done it this way (I also did that in previous alarms) but I am happy to change it if you feel like it would be clearer.

````
resource "aws_cloudwatch_log_metric_filter" "request_temporary_token_api_using_unauthorized_email_address" {
  name           = "RequestTemporaryTokenApiUsingUnauthorizedEmailAddress" <----
  metric_transformation {
    name      = "RequestTemporaryTokenApiUsingUnauthorizedEmailAddress" <----
  }
}
````

Reply:
Good call.  Leave as is.  Cloudwatch Metric Name vs Metric Name in TF
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_metric_filter#name